few options to make clang more compatible with gcc

--- llvm-8.0.0.src/tools/clang/include/clang/Driver/Options.td~	2019-08-26 13:52:15.000000000 +0000
+++ llvm-8.0.0.src/tools/clang/include/clang/Driver/Options.td	2019-08-26 13:54:47.481119265 +0000
@@ -3002,6 +3002,8 @@
 defm branch_count_reg : BooleanFFlag<"branch-count-reg">, Group<clang_ignored_gcc_optimization_f_Group>;
 defm default_inline : BooleanFFlag<"default-inline">, Group<clang_ignored_gcc_optimization_f_Group>;
 defm fat_lto_objects : BooleanFFlag<"fat-lto-objects">, Group<clang_ignored_gcc_optimization_f_Group>;
+defm no_semantic_interposition : BooleanFFlag<"no-semantic-interposition">, Group<clang_ignored_gcc_optimization_f_Group>;
+defm tree_loop_distribute_patterns : BooleanFFlag<"tree-loop-distribute-patterns">, Group<clang_ignored_gcc_optimization_f_Group>;
 defm float_store : BooleanFFlag<"float-store">, Group<clang_ignored_gcc_optimization_f_Group>;
 defm friend_injection : BooleanFFlag<"friend-injection">, Group<clang_ignored_f_Group>;
 defm function_attribute_list : BooleanFFlag<"function-attribute-list">, Group<clang_ignored_f_Group>;
