--- llvm-project-14.0.6.src/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp~	2022-06-22 16:46:24.000000000 +0000
+++ llvm-project-14.0.6.src/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp	2022-08-03 14:54:25.876287446 +0000
@@ -73,7 +73,6 @@
 #include <sys/vt.h>
 #include <linux/cdrom.h>
 #include <linux/fd.h>
-#include <linux/fs.h>
 #include <linux/hdreg.h>
 #include <linux/input.h>
 #include <linux/ioctl.h>
