--- llvm-7.0.0.src/lib/CodeGen/SelectionDAG/DAGCombiner.cpp~	2018-07-30 23:22:00.000000000 +0000
+++ llvm-7.0.0.src/lib/CodeGen/SelectionDAG/DAGCombiner.cpp	2019-04-07 16:11:50.599526633 +0000
@@ -10135,7 +10135,7 @@
 
   SDNodeFlags Flags = N->getFlags();
   bool CanFuse = Options.UnsafeFPMath || isContractable(N);
-  bool AllowFusionGlobally = (Options.AllowFPOpFusion == FPOpFusion::Fast ||
+  bool AllowFusionGlobally = (Options.AllowFPOpFusion == FPOpFusion::Fast || Options.AllowFPOpFusion == FPOpFusion::Standard || 
                               CanFuse || HasFMAD);
   // If the addition is not contractable, do not combine.
   if (!AllowFusionGlobally && !isContractable(N))
