From c06a42b7674cf0c77ac0412e06ec0fe22910e3b7 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Sun, 10 Mar 2019 18:15:11 +0000
Subject: [llvm] Improve physical core count detection

Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
---
 llvm/lib/Support/Host.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git llvm/lib/Support/Host.cpp llvm/lib/Support/Host.cpp
index 36cecf9b2a16..6128f7df1664 100644
--- llvm/lib/Support/Host.cpp
+++ llvm/lib/Support/Host.cpp
@@ -1241,9 +1241,7 @@ int computeHostNumPhysicalCores() {
   llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer>> Text =
       llvm::MemoryBuffer::getFileAsStream("/proc/cpuinfo");
   if (std::error_code EC = Text.getError()) {
-    llvm::errs() << "Can't read "
-                 << "/proc/cpuinfo: " << EC.message() << "\n";
-    return -1;
+    return 1;
   }
   SmallVector<StringRef, 8> strs;
   (*Text)->getBuffer().split(strs, "\n", /*MaxSplit=*/-1,
-- 
2.31.1

