--- llvm-6.0.1.src/lib/Target/X86/X86ISelLowering.h~	2018-05-22 02:46:36.000000000 +0000
+++ llvm-6.0.1.src/lib/Target/X86/X86ISelLowering.h	2018-07-13 16:32:42.450407645 +0000
@@ -829,6 +829,10 @@
     /// Vector-sized comparisons are fast using PCMPEQ + PMOVMSK or PTEST.
     MVT hasFastEqualityCompare(unsigned NumBits) const override;
 
+
+    /// Force aggressive FMA fusion.
+    bool enableAggressiveFMAFusion(EVT VT) const override { return true; }
+
     /// Return the value type to use for ISD::SETCC.
     EVT getSetCCResultType(const DataLayout &DL, LLVMContext &Context,
                            EVT VT) const override;
--- llvm-6.0.1.src/include/llvm/CodeGen/CommandFlags.def~	2017-11-30 13:05:14.000000000 +0000
+++ llvm-6.0.1.src/include/llvm/CodeGen/CommandFlags.def	2018-07-14 02:15:34.760920733 +0000
@@ -171,7 +171,7 @@
 
 static cl::opt<llvm::FPOpFusion::FPOpFusionMode> FuseFPOps(
     "fp-contract", cl::desc("Enable aggressive formation of fused FP ops"),
-    cl::init(FPOpFusion::Standard),
+    cl::init(FPOpFusion::Fast),
     cl::values(
         clEnumValN(FPOpFusion::Fast, "fast", "Fuse FP ops whenever profitable"),
         clEnumValN(FPOpFusion::Standard, "on", "Only fuse 'blessed' FP ops."),
